src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingService.kt

```diff
@@ -299,53 +304,69 @@ class SentencingService(
     eventId: Long,
     request: UpdateCourtAppearanceRequest,
   ) {
-    findPrisoner(offenderNo)
-    findCourtCase(caseId, offenderNo).let { courtCase ->
-      findCourtAppearance(eventId, offenderNo).let { courtAppearance ->
-        courtAppearance.eventDate = request.eventDateTime.toLocalDate()
-        courtAppearance.startTime = request.eventDateTime
-        courtAppearance.courtEventType = lookupMovementReasonType(request.courtEventType)
-        courtAppearance.eventStatus = determineEventStatus(
-          request.eventDateTime.toLocalDate(),
-          courtCase.offenderBooking,
+    findPrisoner(offenderNo).let {
+      findCourtCase(caseId, offenderNo).let { courtCase ->
+        findCourtAppearance(eventId, offenderNo).let { courtAppearance ->
+          courtAppearance.eventDate = request.eventDateTime.toLocalDate()
+          courtAppearance.startTime = request.eventDateTime
+          courtAppearance.courtEventType = lookupMovementReasonType(request.courtEventType)
+          courtAppearance.eventStatus = determineEventStatus(
+            request.eventDateTime.toLocalDate(),
+            courtCase.offenderBooking,
+          )
+          courtAppearance.prison = lookupEstablishment(request.courtId)
+          courtAppearance.outcomeReasonCode =
+            request.outcomeReasonCode?.let { lookupOffenceResultCode(it) }
+          // will get a separate update for a generated next event - so just updating these fields rather than the target appearance
+          courtAppearance.nextEventDate = request.nextEventDateTime?.toLocalDate()
+          courtAppearance.nextEventStartTime = request.nextEventDateTime
+
+          val chargesToUpdateMap = request.courtEventChargesToUpdate.map { it.offenderChargeId to it }.toMap()
+          courtAppearance.courtEventCharges.filter { chargesToUpdateMap.contains(it.id.offenderCharge.id) }
+            .map { courtEventCharge ->
+              val updateCharge = chargesToUpdateMap.getValue(courtEventCharge.id.offenderCharge.id)
+              val resultCode = updateCharge.resultCode1?.let { rs -> lookupOffenceResultCode(rs) }
+              courtEventCharge.offenceDate = updateCharge.offenceDate
+              courtEventCharge.offenceEndDate = updateCharge.offenceEndDate
+              courtEventCharge.mostSeriousFlag = updateCharge.mostSeriousFlag
+              courtEventCharge.offencesCount = updateCharge.offencesCount
+              courtEventCharge.resultCode1 = resultCode
+              courtEventCharge.resultCode1Indicator = resultCode?.dispositionCode
+              with(courtEventCharge.id.offenderCharge) {
+                offenceDate = updateCharge.offenceDate
```

Comment:
so this is the bit we need to look at? Since I guess the NOMIS rules will be only update from latest appearance date